### DISCUSSION TITLE
Coming soon: 4.0.0-beta.2. PR authors, check your release notes!

### DISCUSSION THREAD

**Discussion by ckerr** [transmission/transmission#4251] [Show and tell]
I'm trying out a new script to generate release notes. I've put it up at https://github.com/ckerr/transmission-release-notes for now because it's currently just a kludge. After I clean it up I'll move it to the transmission org.

All maintainers & contributors -- but particularly @mikedld and @Coeur since so many of the PRs are theirs -- should check out these two links:

* https://github.com/transmission/transmission/blob/docs/release-notes/news/news-4.0.0-beta-2.md -- the current draft version
* https://github.com/ckerr/transmission-release-notes/blob/main/README.md#making-better-notes -- how to improve descriptions

There are currently a lot of PRs that currently just read like noise, e.g.: 

* Fix "(arm64) could not find object file symbol for symbol" ([#4031](https://github.com/transmission/transmission/pull/4031))
* macOS: Change BadgeView.mm so that up is up and down is down ([#4055](https://github.com/transmission/transmission/pull/4055))
* Auto enlarge search field when editing ([#4067](https://github.com/transmission/transmission/pull/4067))

I've gone through a lot of the libtransmission PRs but there are a lot more that you two are better qualified to summarize, especially in the macOS client. :green_heart:

**Comment by GaryElshaw** [comment]
Amended: 'macOS: Upload badge above download in Dock icon #4055'

**Comment by ckerr** [comment]
@GaryElshaw here's what the draft looks like: https://github.com/transmission/transmission/blob/docs/release-notes/news/news-4.0.0-beta-2.md and https://github.com/ckerr/transmission-release-notes/blob/main/README.md#making-better-notes has some suggestions on grammar / phrasing.

In this one, for example: 
 - No need to precede with `macOS: ` since the entire section is named `macOS Client` which the script picked up from the `scope:mac` label :tada: 
 - Use past tense and end the sentence with a period, e.g. `Moved the upload badge to be above the download badge in the Dock icon.`

@GaryElshaw also you can annotate other PRs by adding a comment that begins with `Notes: ` to the PR. The next time I run the script it will pick up those annotations.

**Reply by GaryElshaw** [reply]
Done and done.

**Reply by GaryElshaw** [reply]
@Coeur I'd give this a go Antoine, make sure Notes: reflects the change you want to make. Charles hadn't added a note to my pr, so i don't know if you can alter his, but his comment makes me think it's recursive. You can add Notes: and it will be additive.
